src/frontend/src/crypto/ed25519.ts

```diff
@@ -32,13 +32,20 @@ export async function fromSeedWithSlip0010(
  * Create an Ed25519 based on a mnemonic phrase according to SLIP 0010:
  * https://github.com/satoshilabs/slips/blob/master/slip-0010.md
  *
- * The derivation path is an array that is always interpreted as a hardened path.
+ * @param mnemonic A BIP-39 mnemonic.
+ * @param derivationPath an array that is always interpreted as a hardened path.
  * e.g. to generate m/44'/223’/0’/0’/0' the derivation path should be [44, 223, 0, 0, 0]
+ * @param skipValidation if true, validation checks on the mnemonics are skipped.
  */
 export async function fromMnemonic(
   mnemonic: string,
-  derivationPath?: number[]
+  derivationPath: number[] = [],
+  skipValidation = false
 ): Promise<Ed25519KeyIdentity> {
+  if (!skipValidation && !validateMnemonic(mnemonic)) {
+    throw new Error(`Invalid mnemonic: ${mnemonic}`);
```

Comment:
Could we not do this as an exception, but rather as a return value? I don't think this is exceptional behaviour, it's a situation we should expect callers to handle.

The easy way would be to return `Ed25519KeyIdentity | null` unless we see more possible error conditions, in which case we should use a variant.

Reply:
Not that I think about it, we should probably split this into two functions, so the return type on the version that never fails can remain correct. WDYT?

Reply:
Agreed and done.